pkg/keys/ed25519_test.go

```diff
@@ -37,6 +38,32 @@ func (Ed25519Suite) TestUnmarshalEd25519(c *C) {
 	c.Assert(verifier.UnmarshalPublicKey(badKey), IsNil)
 }
 
+func FuzzUnmarshalEd25519(f *testing.F) {
+	f.Fuzz(func(t *testing.T, s string) {
+		if len(s) <= 32 { // ed25519 requires strings of len greater than 32
+			t.Skip()
+		}
+		c := &C{}
+
+		pub, _, err := ed25519.GenerateKey(strings.NewReader(s))
+		c.Assert(err, IsNil)
+
+		publicKey, err := json.Marshal(map[string]string{
+			"public": hex.EncodeToString(pub),
+		})
+		c.Assert(err, IsNil)
+
+		badKey := &data.PublicKey{
+			Type:       data.KeyTypeEd25519,
+			Scheme:     data.KeySchemeEd25519,
+			Algorithms: data.HashAlgorithms,
+			Value:      publicKey,
+		}
+		verifier := NewEd25519Verifier()
+		c.Assert(verifier.UnmarshalPublicKey(badKey), IsNil)
```

Comment:
I'm starting to think that this has uncovered a bug. Shouldn't we see the "unexpected public key length" error? https://github.com/naveensrinivasan/go-tuf/blob/9f957a20ec8019e9bf16028586af7ecbdf9e51b5/pkg/keys/ed25519.go#L64

In general, I don't think every 32-byte string should be a valid ed25519 key (it may not be on the curve). Also, we don't do any crypto in `UnmarshalPublicKey`.